Use Let and Const

spaceninja opened the pull request:
This commit refactors vars to use ES6-style let and const variables.

spaceninja commented:
@vosechu — Mind checking this out? I watched a video on let/const and decided to try updating this code. Seems to work, though of course I'm running the babel-transpiled minified version in the browser.

vosechu commented:
"babel-transpiled minified version in the browser." What has the english language come to?

vosechu commented:
Airbnb has some interesting rules around this: https://github.com/airbnb/javascript#references--prefer-const
Remy seems to prefer the same, and I think he's a smart cookie: https://remysharp.com/2016/08/09/var-const-let
And this is a good clarification for me about the purpose: https://mathiasbynens.be/notes/es6-const

So I'm going to approve now. Seems like you got it right :)

vosechu commented:
Hah, a counter point opinion: http://blog.getify.com/constantly-confusing-const/#premature-const

spaceninja commented:
Thank @vosechu!

* I ditched Babel and am now running the raw ES6 code in browser. Since I'm learning and this is a side project, no reason to add a layer of abstraction. This does mean the .min.js file isn't actually minified anymore, since the version of UglifyJS included in CodeKit doesn't grok ES6, but oh well, it's a small file.

* I switched to ESLint for linting, which understands ES6 better, and turned on a few of the settings that AirBnB styleguide recommended.

* I added `use strict` to each function the linter complained about. I'm a little confused why it doesn't like me putting it in the global space, but 🤷‍♂️ 

* I changed one `let` to `const` because it wasn't being reassigned, and the linter didn't like it. I see the point that guy made about prematurely optimizing, but in this case, I'd rather follow some linter best practices and potentially have some refactoring in the future than let myself get into bad habits now.

Anyways, should be good now if you don't mind taking another look.

vosechu commented on js/listdates-min.js:
```diff
@@ -1 +1,81 @@
+};
+
+// Send the Ajax request
+xhr.send();
```
Non-blocking: Any reason to even keep this?

spaceninja replied:
I'm not sure what you mean? The comment? I suppose not, but I was kinda deliberately over-commenting as a learning technique

vosechu commented:
https://addyosmani.com/blog/essential-js-namespacing/#beginners

This is the best explanation I've found so far. And Addy is amazing.

spaceninja commented:
Ah, that makes more sense if the standard pattern is to wrap in an IIFE, then you only need a single `use strict` — I've got a ticket open for that, I'll look into doing that next.
